compiler/rustc_middle/src/ty/fast_reject.rs

```diff
@@ -60,29 +60,23 @@ pub enum StripReferences {
     No,
 }
 
-/// Tries to simplify a type by only returning the outermost injective¹ layer, if one exists.
+/// Tries to simplify a type by only returning the outermost layer of an inductive type, if one exists.
 ///
 /// The idea is to get something simple that we can use to quickly decide if two types could unify,
 /// for example during method lookup.
 ///
-/// A special case here are parameters and projections. Projections can be normalized to
-/// a different type, meaning that `<T as Trait>::Assoc` and `u8` can be unified, even though
-/// their outermost layer is different while parameters like `T` of impls are later replaced
-/// with an inference variable, which then also allows unification with other types.
+/// We have to be careful with parameters and projections as we might want them to unify with other types.
+/// Projections can be normalized to a different type, meaning that `<T as Trait>::Assoc`
+/// and `u8` can be unified, even though their outermost layer is different while parameters like `T` of
+/// impls are later replaced with an inference variable, which then also allows unification with other types.
 ///
-/// When using `SimplifyParams::Yes`, we still return a simplified type for params and projections²,
-/// the reasoning for this can be seen at the places doing this.
+/// When using `SimplifyParams::Yes`, we still return a simplified type for params.
```

Comment:
I wonder if a better name for this would be `TreatParamsAsPlaceholders`-- I've been trying to push the "placeholder" terminology for "universally bound free variable".

Reply:
Or something like 

```rust
enum TreatParams {
    AsPlaceholderTypes,
    AsBoundTypes,
}
```